lib/internal/mime.js

```diff
@@ -16,62 +16,140 @@ const {
   ERR_INVALID_MIME_SYNTAX,
 } = require('internal/errors').codes;
 
-const NOT_HTTP_TOKEN_CODE_POINT = /[^!#$%&'*+\-.^_`|~A-Za-z0-9]/g;
-const NOT_HTTP_QUOTED_STRING_CODE_POINT = /[^\t\u0020-~\u0080-\u00FF]/g;
-
-const END_BEGINNING_WHITESPACE = /[^\r\n\t ]|$/;
-const START_ENDING_WHITESPACE = /[\r\n\t ]*$/;
+const NOT_HTTP_TOKEN_CODE_POINT = /[^!#$%&'*+\-.^_`|~A-Za-z0-9]/;
+const NOT_HTTP_QUOTED_STRING_CODE_POINT = /[^\t\u0020-~\u0080-\u00FF]/;
 
+const SOLIDUS = '/';
+const WHITESPACE_REGEX = /[^\r\n\t ]/;
+const SEMICOLON = ';';
+const BACKSLASH = '\\';
+const QUOTE = '"';
+const ASCII_LOWER_LOWERBOUND = StringPrototypeCharCodeAt('A');
+const ASCII_LOWER_UPPERBOUND = StringPrototypeCharCodeAt('Z');
+const SEMICOLON_CHARCODE = StringPrototypeCharCodeAt(SEMICOLON);
+const EQUALS_CHARCODE = StringPrototypeCharCodeAt('=');
+const BACKSLASH_CHARCODE = StringPrototypeCharCodeAt(BACKSLASH);
+const QUOTE_CHARCODE = StringPrototypeCharCodeAt(QUOTE);
 function toASCIILower(str) {
   let result = '';
   for (let i = 0; i < str.length; i++) {
     const char = str[i];
+    const charCode = StringPrototypeCharCodeAt(char, 0);
 
-    result += char >= 'A' && char <= 'Z' ?
+    result += charCode >= ASCII_LOWER_LOWERBOUND &&
+      charCode <= ASCII_LOWER_UPPERBOUND ?
       StringPrototypeToLowerCase(char) :
       char;
   }
   return result;
 }
 
-const SOLIDUS = '/';
-const SEMICOLON = ';';
+// https://fetch.spec.whatwg.org/#collect-an-http-quoted-string
+function collectHTTPQuotedString(str, marker) {
+  const limit = str.length;
+  let index = marker.position;
+  // #2 Let value be the empty string.
+  let value = '';
+
+  // #4 Advance position by 1.
+  index++;
+
+  // #5 While true
+  while (true) {
+    let nextDelimiter = StringPrototypeIndexOf(str, BACKSLASH, index);
+
+    nextDelimiter = nextDelimiter !== -1 ? nextDelimiter : StringPrototypeIndexOf(str, QUOTE, index);
+
+    // #5-1 Append the result of collecting a sequence of code points
+    // that are not U+0022 (") or U+005C (\) from input, given position, to value.
+    const collectedPoints = nextDelimiter === -1 ?
+      StringPrototypeSlice(str, index) : StringPrototypeSlice(str, index, nextDelimiter);
+    index += collectedPoints.length;
+    value += collectedPoints;
+
+    // #5-2 If position is past the end of input, then break.
+    if (index > limit) break;
+
+    // #5-3 Let quoteOrBackslash be the code point at position within input.
+    const quoteOrBackslash = str[index];
+    const quoteOrBackslashAsCode = quoteOrBackslash == null ?
+      0 :
+      StringPrototypeCharCodeAt(quoteOrBackslash, 0);
+
+    // #5-4 Advance position by 1.
+    index++;
```

Comment:
check if there is perf difference by doing 

```suggestion
    ++index;
```